### PR TITLE
Added NAN validation

### PR DESCRIPTION
**Pull request by iamraccoon**
This adds the ability to validate NAN values.

### CODE REVIEW

File: library/Rules/NanType.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+class NanType extends AbstractRule
+{
+    public function validate($input)
+    {
+        if (is_numeric($input)) {
+            return is_nan($input);
+        }
+
+        throw new ComponentException('The value must be a numeric.');
```
Comment (henriquemoody) [inline library/Rules/NanType.php]: The `validate()` method doesn't throw exceptions.

**Comment by henriquemoody** [comment]
Why not use `v::not(v::numeric())`?

**Comment by henriquemoody** [comment]
Actually after taking a close look at this I realized what it really means. A _numeric_ value may also be an unrepresentable number, that's why `is_nan()` is for. 

However, this rule would be more useful if called _Number_, if you revert the operation. Can you provide this change among with the other request?

**Comment by wcbrown11** [comment]
Please post a job for free at careerduh.byethost31.com/

Skip the payment form.

Thank you,
William - manager 
CareerDuh

**Comment by henriquemoody** [comment]
Thanks for contributing! 🐼

**Pull request closed without merging**